jwst/regtest/test_NIRSpec_IRS2_detector1.py

```diff
@@ -0,0 +1,86 @@
+
+    Step.from_cmdline(["jwst.pipeline.Detector1Pipeline", rtdata.input,
+                       "--save_calibrated_ramp=True",
+                       "--steps.group_scale.save_results=True",
+                       "--steps.dq_init.save_results=True",
+                       "--steps.saturation.save_results=True",
+                       "--steps.ipc.save_results=True",
+                       "--steps.superbias.save_results=True",
+                       "--steps.refpix.save_results=True",
+                       "--steps.rscd.save_results=True",
+                       "--steps.linearity.save_results=True",
+                       "--steps.dark_current.save_results=True",
+                       "--steps.jump.save_results=True",
+                       "--steps.ramp_fit.save_results=True",
+                       "--steps.gain_scale.save_results=True",
+                       "--steps.jump.rejection_threshold=20.0"])
+
+    return rtdata
+
+@pytest.mark.bigdata
+def test_NIRSpec_det1_completion(run_pipeline):
+    """ Check that the final and intermediate files are generated by the pipeline."""
+    files = glob('*_rate.fits')
+    files += glob('*_ramp.fits')
+    files += glob('*_dark_current.fits')
+    files += glob('*_dq_init.fits')
+    files += glob('*_gain_scale.fits')
+    files += glob('*_group_scale.fits')
+    files += glob('*_ipc.fits')
+    files += glob('*_jump.fits')
+    files += glob('*_linearity.fits')
+    files += glob('*_refpix.fits')
+    files += glob('*_saturation.fits')
+    files += glob('*_superbias.fits')
+    files += glob('*_0_ramp_fit.fits')
+    # There should be 13 outputs
+    assert len(files) == 13
+
+
+@pytest.mark.bigdata
+@pytest.mark.parametrize("output", [
+    'jw0010010_11010_nrs1_chimera_rate.fits',
+    'jw0010010_11010_nrs1_chimera_ramp.fits',
+    'jw0010010_11010_nrs1_chimera_dark_current.fits',
+    'jw0010010_11010_nrs1_chimera_dq_init.fits',
+    'jw0010010_11010_nrs1_chimera_gain_scale.fits',
+    'jw0010010_11010_nrs1_chimera_group_scale.fits',
+    'jw0010010_11010_nrs1_chimera_ipc.fits',
+    'jw0010010_11010_nrs1_chimera_jump.fits',
+    'jw0010010_11010_nrs1_chimera_linearity.fits',
+    'jw0010010_11010_nrs1_chimera_refpix.fits',
+    'jw0010010_11010_nrs1_chimera_saturation.fits',
+    'jw0010010_11010_nrs1_chimera_superbias.fits',
+    'jw0010010_11010_nrs1_chimera_0_ramp_fit.fits',],
+                         ids=['rate','ramp','dark_current','dq_init','gain_scale',
+                              'group_scale','ipc','jump','linearity','refpix',
+                              'saturation','superbias','0_ramp_fit'])
+def test_NIRSpec_det1Pipeline(run_pipeline, fitsdiff_default_kwargs, output):
+    """
+    Regression test of calwebb_Detector1 pipeline performed on NIRSpec data.
+    """
+    rtdata = run_pipeline
+    rtdata.output = output
+    rtdata.get_truth(os.path.join("truth/nirspec/test_detector1Pipeline", output))
```

Comment:
The path should be "truth/test_nirspec_irs2_detector1/", though I'm not sure whether the directory below "truth/" should be lower case or match the case of the test name.

Reply:
Test names should be lowercase as it is a test function.  Functions in python are always lower case with underscores.  Only classes get CamelCase, but without underscores.  Never mix.  Same with this test filename - should be lower case.